pkg/rekor/identity.go

```diff
@@ -351,25 +371,55 @@ func writeIdentitiesBetweenCheckpoints(logInfo *models.LogInfo, prevCheckpoint *
 	startIndex := int(prevCheckpoint.Size) + totalSize - 1 //nolint: gosec // G115, log will never be large enough to overflow
 	endIndex := int(checkpoint.Size) + totalSize - 1       //nolint: gosec // G115
 
-	// Search for identities in the log range
-	if identity.MonitoredValuesExist(monitoredValues) {
-		entries, err := GetEntriesByIndexRange(context.Background(), rekorClient, startIndex, endIndex)
-		if err != nil {
-			return fmt.Errorf("error getting entries by index range: %v", err)
+	return startIndex, endIndex
+}
+
+func IdentitySearch(startIndex int, endIndex int, rekorClient *client.Rekor, monitoredValues identity.MonitoredValues, outputIdentitiesFile string, idMetadataFile *string) error {
+
+	entries, err := GetEntriesByIndexRange(context.Background(), rekorClient, startIndex, endIndex)
+	if err != nil {
+		return fmt.Errorf("error getting entries by index range: %v", err)
+	}
+	idEntries, err := MatchedIndices(entries, monitoredValues)
+	if err != nil {
+		return fmt.Errorf("error finding log indices: %v", err)
+	}
+
+	if len(idEntries) > 0 {
+		for _, idEntry := range idEntries {
+			fmt.Fprintf(os.Stderr, "Found %s\n", idEntry.String())
+
+			if err := file.WriteIdentity(outputIdentitiesFile, idEntry); err != nil {
+				return fmt.Errorf("failed to write entry: %v", err)
+			}
 		}
-		idEntries, err := MatchedIndices(entries, monitoredValues)
+	}
+
+	// TODO: idMetadataFile currently takes in a string pointer to not cause a regression in the current reusable monitoring workflow.
+	// Once the reusable monitoring workflow is split into a consistency check and identity search, idMetadataFile should always take in a string value.
```

Comment:
Potentially could also leave as an optional in case users don't want the extra file write/use case of identity metadata file